pwnlib/tubes/process.py

```diff
@@ -667,6 +667,40 @@ def kill(self):
         """
         self.close()
 
+    def terminate(self):
+        """terminate()
+
+        Terminates the process by sending SIGTERM.
+        
+        This is a more graceful way to stop a process compared to kill(),
+        which sends SIGKILL. The process has a chance to clean up and
+        exit gracefully when receiving SIGTERM.
+        
+        Examples:
+        
+            >>> p = process(['sleep', '10'])
+            >>> p.terminate()
+            >>> p.poll() is not None
+            True
+        """
+        if self.proc is None:
+            return
+            
+        # Check if process is still running
+        if self.poll() is not None:
+            return
+            
+        try:
+            if IS_WINDOWS:
+                # On Windows, terminate() is equivalent to kill()
+                self.proc.terminate()
+            else:
+                # On Unix-like systems, send SIGTERM
+                os.kill(self.pid, signal.SIGTERM)
```

Comment:
I don't think we need this distinction since `os.terminate` is documented to send SIGTERM on posix-like systems. So we can just call `terminate` disregarding the host os